src/wms/src/main/java/org/geoserver/wms/NearestMatchWarningAppender.java

```diff
@@ -0,0 +1,85 @@
+/* (c) 2018 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms;
+
+import org.geoserver.ows.AbstractDispatcherCallback;
+import org.geoserver.ows.Request;
+import org.geoserver.ows.Response;
+import org.geoserver.platform.Operation;
+import org.springframework.http.HttpHeaders;
+
+import javax.servlet.http.HttpServletResponse;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * Appends warning messages in case of nearest match
+ */
+public class NearestMatchWarningAppender extends AbstractDispatcherCallback {
+
+    static final TimeZone UTC_TZ = TimeZone.getTimeZone("UTC");
+
+    public enum WarningType {Nearest, Default, NotFound}
+
+    static final ThreadLocal<List<String>> WARNINGS = ThreadLocal.withInitial(() -> new ArrayList<>());
+
+    /**
+     * Adds a default value or nearest value warning.
+     *
+     * @param layerName   Mandatory, the layer being used
+     * @param dimension   The dimension name
+     * @param value       The actual value used, or null if it's a {@link WarningType#NotFound} warning type
+     * @param unit        The measure unit of measure
+     * @param warningType
+     */
+    public static void addWarning(String layerName, String dimension, Object value, String unit, WarningType 
+            warningType) {
+        List<String> warnings = WARNINGS.get();
+        if (warningType == WarningType.NotFound) {
+            warnings.add("99 No nearest value found on " + layerName + ": " + dimension);
```

Comment:
99 ?

Reply:
Yes, by specification. Copying verbatim:

In order that the client may determine what value was actually sent, the server shall label the response object with the actual value found by rounding. In the HTTP environment, the labelling is performed using an HTTP response header. For each dimension to which a rounded-off value was applied, a header line of the following form shall be sent:
``Warning: 99 Nearest value used: DIM_NAME=value units``
where "99" is a defined by HTTP (IETF RFC 2616) for use by miscellaneous warnings, DIM_NAME is the corresponding request parameter name, value is the value actually used, and units is the units attribute (see C.2) for that dimension.